### PR TITLE
Test directory sign compare

### PR DESCRIPTION
**Pull request by cvvergara**
#4018 

## Tasklist

- [x] fix -Wsign-compare
- [x] add line on changelog (not needed)

substitutions:
int -> size_t
int -> rapidjson::SizeType

some fixes use the:
```
uint32_t invalid<uint32_t>(); 
....
.. is_valid(kmh) ...
```

On the file with lambdas:
```
LiveTrafficCustomize close_edge = [] .... ( int index -> unit32_t index, ...)
```

**Note:**
 
test/http_tiles.cc contains:
```
#include "valhalla/filesystem.h"  
```
By modifying the file `test/http_tiles.cc` the following warning is expected when running `scripts/clang-tidy-only-diff.sh 4`
```
/home/circleci/project/valhalla/filesystem.h:214:7: warning: Call to function 'strcpy' is insecure as it does not provide bounding of the memory buffer. Replace unbounded copy functions with analogous functions that support length arguments such as 'strlcpy'. CWE-119 [clang-analyzer-security.insecureAPI.strcpy]
      strcpy(entry_->d_name, filename.c_str());
      ^
/home/circleci/project/valhalla/filesystem.h:214:7: note: Call to function 'strcpy' is insecure as it does not provide bounding of the memory buffer. Replace unbounded copy functions with analogous functions that support length arguments such as 'strlcpy'. CWE-119
```

### CODE REVIEW

File: test/gurka/test_osrm_serializer.cc [hunk 1]
```diff
@@ -146,9 +146,9 @@ TEST(Standalone, HeadingNumberCrossRoad) {
 
   auto json = gurka::convert_to_json(result, Options::Format::Options_Format_osrm);
   std::vector<int> expected_headings{1, 4, 1};
-  for (int i = 0; i < json["routes"][0]["legs"][0]["steps"].GetArray().Size(); ++i) {
-    for (int j = 0; j < json["routes"][0]["legs"][0]["steps"][i]["intersections"].GetArray().Size();
-         ++j) {
+  for (size_t i = 0; i < json["routes"][0]["legs"][0]["steps"].GetArray().Size(); ++i) {
```
Comment (kevinkreiser) [inline test/gurka/test_osrm_serializer.cc:149]: i think this is actually `unsigned` vs `size_t` which means the `operator[]` will narrow it again

https://github.com/Tencent/rapidjson/blob/master/include/rapidjson/rapidjson.h#L415

File: test/shape_attributes.cc [hunk 2]
```diff
@@ -59,7 +59,7 @@ TEST(ShapeAttributes, test_shape_attributes_included) {
 
   // Assert that the shape attributes (time, length, speed) are equal to their corresponding edge
   // attributes
-  for (int e = 0; e < edges.Size(); e++) {
+  for (rapidjson::SizeType e = 0; e < edges.Size(); e++) {
```
Comment (kevinkreiser) [inline test/shape_attributes.cc:62]: ah strange you used it here!

**Review by kevinkreiser — DISMISSED** [review]
looks good, did you want to get those other instances of `rapidjson::SizeType`?

**Review by nilsnolde — COMMENTED** [review]
Since we're closing #4226, please refer to the comments in there, otherwise it's fine for me.

File: test/gurka/test_osrm_serializer.cc [hunk 3]
```diff
@@ -53,11 +53,12 @@ TEST(Standalone, HeadingNumberTurnLeft) {
                                  {{"/shape_format", "geojson"}});
   //
   auto json = gurka::convert_to_json(result, Options::Format::Options_Format_osrm);
-  auto heading_number{1};
-  for (int i = 0; i < json["routes"][0]["legs"][0]["steps"].GetArray().Size(); ++i)
-    ASSERT_EQ(heading_number, json["routes"][0]["legs"][0]["steps"][0]["intersections"][0]["bearings"]
-                                  .GetArray()
-                                  .Size());
+  std::vector<int> expected_headings{1, 2, 1};
```
Comment (cvvergara) [inline test/gurka/test_osrm_serializer.cc:56]: Based on the other tests, I also modified this cycle to test all the expected headings.
Now, variable `i` in the cycle is been used

File: test/shape_attributes.cc (hunk 2 above)
Comment (cvvergara) [reply test/shape_attributes.cc:62]: done

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**